src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -294,9 +294,12 @@ private void LogTelemetryForLegacySettings(IRequestData requestData, string runs
         {
             requestData.MetricsCollection.Add(TelemetryDataConstants.TestSettingsUsed, InferRunSettingsHelper.IsTestSettingsEnabled(runsettings));
             
-            if (InferRunSettingsHelper.TryGetLegacySettingElements(runsettings, out List<string> legacySettings))
+            if (InferRunSettingsHelper.TryGetLegacySettingElements(runsettings, out Dictionary<string, string> legacySettingsCIData))
             {
-                requestData.MetricsCollection.Add(TelemetryDataConstants.LegacySettingElements, string.Join(",", legacySettings));
+                foreach( var ciData in legacySettingsCIData)
+                {
```

Comment:
Please put a comment here, example for what is being logged

Reply:
Done

Reply:
Also add an example,
key = VS.TestRun.LegacySettings.Elements, value = "Deployment, Scripts, Execution"